### PR TITLE
Rendertext output

### PR DESCRIPTION
**Pull request by marton-balazs-kovacs**
Hi Frederik,

I have changed the two text output functions to format text as rmd (for download), HTML (for preview), raw (for clipping). I also modified the corresponding modules. Now it is faster on the preview. I was not sure whether we should make this modification or not so I created this crude version. I know the print function codes are not efficient, as you have to modify the output text three times right now.

Please take a look at it and let me know whether you think it is a good way to go. Or there is another structure or solution we should use, or leave it as it is now. I really do not know.

Best,
Marton

### CODE REVIEW

File: R/print_contrib_affil.R [hunk 1]
```diff
@@ -41,32 +41,75 @@ print_contrib_affil <- function(infosheet) {
...
+  
+  if (output_format == "rmd") {
+    contrib_print <-
+      contrib_data %>% 
+      dplyr::transmute(contrib = purrr::pmap_chr(list(Names, `Primary affiliation`, `Secondary affiliation`),
+                                                 ~ dplyr::if_else(is.na(..3),
+                                                                  glue::glue("{..1}^{..2}^"),
+                                                                  glue::glue("{..1}^{..2},{..3}^")))) %>% 
+      dplyr::pull(contrib) %>% 
+      glue::glue_collapse(., sep = ", ")
```
Comment (crsh) [inline R/print_contrib_affil.R:57]: To reduce redundancies a little these two lines could be applied after the if-else-block, no?

File: R/print_contrib_affil.R [hunk 2]
```diff
@@ -41,32 +41,75 @@ print_contrib_affil <- function(infosheet) {
...
-    dplyr::pull(affil) %>% 
-    glue::glue_collapse(., sep = ", ")
+    dplyr::distinct(affiliation, .keep_all = TRUE)
+  
+  if (output_format == "rmd") {
+    affil_print <-
+      affil_data %>% 
+      dplyr::transmute(affil = glue::glue("^{affiliation_no}^{affiliation}")) %>% 
+      dplyr::pull(affil) %>% 
+      glue::glue_collapse(., sep = ", ")
```
Comment (crsh) [inline R/print_contrib_affil.R:90]: Again, to reduce redundancies a little these two lines could be applied after the if-else-block, no?

File: R/print_roles_readable.R [hunk 3]
```diff
@@ -33,8 +34,22 @@ print_roles_readable <-  function(infosheet) {
...
-    dplyr::pull(out)
-}
\ No newline at end of file
+    dplyr::summarise(Names = glue::glue_collapse(Name, sep = ", ", last = " and "))
+  
+  if (output_format == 'rmd') {
+    res %>% 
+      dplyr::transmute(out = glue::glue("**{`CRediT Taxonomy`}:** {Names}.")) %>% 
+      dplyr::summarise(out = glue::glue_collapse(out, sep = "  \n")) %>% 
+      dplyr::pull(out)
```
Comment (crsh) [inline R/print_roles_readable.R:43]: Here, it's just one line, but this could also be done after the if-else-block.

**Comment by marton-balazs-kovacs** [comment]
Thanks for the review.
I will place the redundant rows outside of the if-else block. Later on, we can restructure the code to make it nicer. If you have any ideas about this please let me know.

**Comment by crsh** [comment]
Looks good to me. Nice work.

**Pull request merged**